optimade/server/mappers/entries.py

```diff
@@ -88,37 +91,103 @@ def length_alias_for(cls, field: str) -> Optional[str]:
+
+        Returns:
+            The mapped field name to be used in the query to the backend.
+
+        """
+        split = optimade_field.split(".")
+        alias = dict(cls.all_aliases()).get(split[0], None)
+        if alias is not None:
+            return alias + ("." + ".".join(split[1:]) if len(split) > 1 else "")
+        return optimade_field
+
     @classmethod
     def alias_for(cls, field: str) -> str:
         """Return aliased field name.
 
+        !! warning: This method is deprecated could be removed without further warning. Please
+        use [`get_backend_field()`](optimade.server.mappers.entries.BaseResourceMapper.get_backend_field].
+
         Parameters:
             field: OPTIMADE field name.
 
         Returns:
             Aliased field as found in [`all_aliases()`][optimade.server.mappers.entries.BaseResourceMapper.all_aliases].
 
         """
-        split = field.split(".")
-        alias = dict(cls.all_aliases()).get(split[0], None)
-        if alias is not None:
-            return alias + ("." + ".".join(split[1:]) if len(split) > 1 else "")
-        return field
+        warnings.warn(
+            "The `.alias_for(...)` method is deprecated, please use `.get_backend_field(...)`.",
+            DeprecationWarning,
+        )
+        return cls.get_backend_field(field)
+
+    @classmethod
+    def get_optimade_field(cls, backend_field: str) -> str:
+        """Return the corresponding OPTIMADE field name for the underlying database field,
+        ready to be used to construct the OPTIMADE-compliant JSON response.
+
+        Aliases are read from
+        [`all_aliases()`][optimade.server.mappers.entries.BaseResourceMapper.all_aliases].
+
+        Arguments:
+            backend_field: The backend field to attempt to map to an OPTIMADE field.
+
+        Examples:
+            >>> get_backend_field("chemical_formula_anonymous")
+            'chemical_formula_anonymous'
+            >>> get_backend_field("formula_anon")
+            'chemical_formula_anonymous'
+            >>> get_backend_field("custom_provider_field")
+            '_exmpl_custom_provider_field'
+
+        Returns:
+            The mapped field name to be used in an OPTIMADE-compliant response.
+
+        """
+        return {alias: real for real, alias in cls.all_aliases()}.get(
+            backend_field, backend_field
+        )
 
     @classmethod
     def alias_of(cls, field: str) -> str:
         """Return de-aliased field name, if it exists,
         otherwise return the input field name.
 
-        Args:
+        !! warning: This method is deprecated could be removed without further warning. Please
+        use [`get_optimade_field()`](optimade.server.mappers.entries.BaseResourceMapper.get_optimade_field].
+
+        Parameters:
             field: Field name to be de-aliased.
 
         Returns:
             De-aliased field name, falling back to returning `field`.
 
         """
-        field = field.split(".")[0]
```

Comment:
I removed this line from the "new" method as it is not tested anywhere and therefore should not be relied on for anything; I think it may just be left over from when `alias_for` was modified to make this method. If you ask for the OPTIMADE field of e.g. `my_weird_field.with_dots`, you would want it to return `my_weird_field.with_dots` rather than `alias_for_my_weird_field` even if the dot has some significance for that particular backend.